CMakeLists.txt

```diff
@@ -1,61 +1,44 @@
 # ensmallen CMake configuration.  This project has no configurable options---it
 # just installs the headers to the install location, and optionally builds the
 # test program.
-cmake_minimum_required(VERSION 2.8.10)
-project(ensmallen C CXX)
+cmake_minimum_required(VERSION 3.3.2)
+project(ensmallen 
+        VERSION 2.10.5
```

Comment:
Do we need to specify `VERSION` here?  (Does it get us anything?)  I can see this going out of date very quickly if we're not careful.  In fact technically it's already out of date---2.11.0 is the newest version, but the git master branch should probably have a different version number.  So if we can avoid adding this I think it might be better; but I am not familiar with the `VERSION` option, so maybe there is something I don't know about why it's needed.

Reply:
It probably wasn't clear because I hadn't written the code for exporting the target, it was a misunderstanding on my part. The `VERSION` option helps us create the version config file, which is what another downstream project would use to check whether they are importing a compatible version, if they used `find_package(ensmallen 2.10.0)`. You can see this in the wiki [here](https://gitlab.kitware.com/cmake/community/wikis/doc/tutorials/Packaging).

Reply:
Instead of hardcoding the version number into CMakeLists.txt, it's possible to extract the version of ensmallen directly from the C++ source file.  Example: https://gitlab.com/conradsnicta/armadillo-code/blob/9.800.x/CMakeLists.txt#L68-81

Reply:
I didn't know we had this file, thanks :)

Reply:
Got it, thanks for the clarification. :+1: